deno/src/main.ts

```diff
@@ -0,0 +1,204 @@
+    if (!fileMeta) continue;
+    const [checksumType, checksumValue] = splitOnce(fileMeta.checksum, "-");
+
+    const url = `https://jsr.io/${pkg.module}/${pkg.version}${fileUrl}`;
+    let [fileDir, fileName] = splitOnce(fileUrl, "/", "right");
+    const dest = `vendor/jsr.io/${pkg.module}/${pkg.version}${fileDir}`;
+
+    if (shouldHash(fileName)) {
+      fileName = await shortHash(fileName);
+    }
+
+    flatpkData.push({
+      type: "file",
+      url,
+      [checksumType]: checksumValue,
+      dest,
+      "dest-filename": fileName,
+    });
+  }
+
+  // If a moule imports deno.json (import ... from "deno.json" with {type:"json"}), it won't appear in the module graph
+  // Worarkound: if there is a deno.json file in the manifest just add it
+  // Note this can be made better, by looking in the moduleGraph if deno.json is specified in the dependencies
+  for (const [fileUrl, fileMeta] of Object.entries(metaVer.manifest)) {
+    if (fileUrl.includes("deno.json")) {
+      const [checksumType, checksumValue] = splitOnce(
+        // deno-lint-ignore no-explicit-any
+        (fileMeta as any).checksum,
+        "-",
+      );
+      const url = `https://jsr.io/${pkg.module}/${pkg.version}${fileUrl}`;
+      const [fileDir, fileName] = splitOnce(fileUrl, "/", "right");
+      const dest = `vendor/jsr.io/${pkg.module}/${pkg.version}${fileDir}`;
+      flatpkData.push({
+        type: "file",
+        url,
+        [checksumType]: checksumValue,
+        dest,
+        "dest-filename": fileName,
+      });
+    }
+  }
+
+  return flatpkData;
+}
+
+async function npmPkgToFlatpakData(pkg: Pkg) {
+  //url: https://registry.npmjs.org/@napi-rs/cli/-/cli-2.18.4.tgz
+  //npmPkgs;
+  const metaUrl = `https://registry.npmjs.org/${pkg.module}`;
+  const metaText = await fetch(metaUrl).then(
+    (r) => r.text(),
+  );
+  const meta = JSON.parse(metaText);
+
+  const metaData = {
+    type: "file",
+    url: metaUrl,
+    sha256: await sha256(metaText),
+    dest: `deno_dir/npm/registry.npmjs.org/${pkg.module}`,
+    "dest-filename": "registry.json",
+  };
+
+  const [checksumType, checksumValue] = splitOnce(
+    meta.versions[pkg.version].dist.integrity,
+    "-",
+  );
+  const pkgData: Record<string, unknown> = {
+    type: "archive",
+    "archive-type": "tar-gzip",
+    url:
+      `https://registry.npmjs.org/${pkg.module}/-/${pkg.name}-${pkg.version}.tgz`,
+    [checksumType]: base64ToHex(checksumValue),
+    dest: `deno_dir/npm/registry.npmjs.org/${pkg.module}/${pkg.version}`,
+  };
+
+  if (pkg.cpu) {
+    pkgData["only-arches"] = [pkg.cpu];
+  }
+
+  return [metaData, pkgData];
+}
+
+if (import.meta.main) {
+  const arg = Deno.args[0];
+  if (!arg) {
+    console.error("No argument provided");
+    Deno.exit(1);
+  }
+
+  const lock = JSON.parse(Deno.readTextFileSync(arg));
+  if (lock.version !== "5") {
+    throw new Error(`Unsupported deno lock version: ${lock.version}`);
+  }
```

Comment:
this should be documented in readme. I assume lock version 5 requires a particular version of deno.